src/components/Contracts/ContractAgreementsTable.js

```diff
@@ -0,0 +1,328 @@
+    setContracts(results);
+    setTotalContracts(totalCount);
+
+    setIsLoading(false); // hide loading indicator when data is fetched
+  }
+
+  async function getContractAgreements(fetchAll = false) {
+    const filtersToSubmit = { ...filter };
+
+    console.log('getContractAgreements before ', filtersToSubmit);
+    // filter out keys we don't want to submit
+    Object.keys(filtersToSubmit).forEach((k) => {
+      if (k === 'organization_id' && filtersToSubmit[k].length) {
+        filtersToSubmit['growing_organization_id'] = filtersToSubmit[k];
+        delete filtersToSubmit[k];
+        delete filtersToSubmit['sub_organization'];
+      }
+      if (
+        filtersToSubmit[k] === 'all' ||
+        filtersToSubmit[k] === '' ||
+        filtersToSubmit[k] === undefined
+      ) {
+        delete filtersToSubmit[k];
+      }
+    });
+
+    const queryParams = {
+      offset: fetchAll ? 0 : page * contractsPerPage,
+      limit: fetchAll ? 90000 : contractsPerPage,
+      // sort_by: sortBy?.field,
+      // order: sortBy?.order,
+      ...filtersToSubmit,
+    };
+
+    console.log('getContractAgreements after', queryParams);
+
+    // log.debug('queryParams', queryParams);
+
+    const response = await contractsAPI.getContractAgreements(queryParams);
+    console.log('getContractAgreements response: ', response);
+
+    const results = prepareRows(response.agreements);
+    return {
+      results,
+      totalCount: response.query.count,
+    };
+  }
+
+  const handleOpenMainFilter = () => setIsMainFilterOpen(true);
+  const handleOpenDateFilter = () => setIsDateFilterOpen(true);
+
+  useEffect(() => {
+    // console.log('contractAgreementsTable usEffect filter', filter);
+    if (filter?.start_date && filter?.end_date) {
+      const dateRangeString = generateActiveDateRangeFilterString(
+        filter?.start_date,
+        filter?.end_date
+      );
+      setActiveDateRangeString(dateRangeString);
+    } else {
+      setActiveDateRangeString('');
+    }
+
+    getContracts();
+  }, [page, contractsPerPage, /*sortBy,*/ filter]);
+
+  return (
+    <>
+      <div className={classes.flex}>
+        <CreateContractAgreement />
+      </div>
+
+      <CustomTable
+        setPage={setPage}
+        page={page}
+        sortBy={sortBy}
+        rows={contracts}
+        isLoading={isLoading}
+        activeDateRange={activeDateRangeString}
+        setRowsPerPage={setContractsPerPage}
+        rowsPerPage={contractsPerPage}
+        setSortBy={setSortBy}
+        totalCount={totalContracts}
+        openMainFilter={handleOpenMainFilter}
+        openDateFilter={handleOpenDateFilter}
+        handleGetData={getContracts}
+        setSelectedRow={(value) => {
+          setSelectedContractAgreement(value);
+          setDetailShown(true);
+        }}
+        selectedRow={selectedContractAgreement}
+        tableMetaData={contractsTableMetaData}
+        activeFiltersCount={
```

Comment:
Pressing Reset does not clear the filter count and doesn't correctly reset the table. This may be a bug in the common component though.